binaries/daemon/src/lib.rs

```diff
@@ -114,6 +114,8 @@ pub struct Daemon {
     sessions: BTreeMap<SessionId, BuildId>,
     builds: BTreeMap<BuildId, BuildInfo>,
     git_manager: GitManager,
+    /// System instance for metrics collection (reused across calls)
+    metrics_system: Option<sysinfo::System>,
```

Comment:
Why is this an `Option`? Is there any way how this can be `None`? If not, let's remove the option so that we can avoid the potential error case.